### PR TITLE
Clean up graphics HOBs

### PR DESCRIPTION
**Pull request by gdong1**
Removing the HOB defined by SBL to use generic HOBs from EDK2.
Update GraphicsLib for above HOB change.
Update osloader and multibootLib for hob change.
update APL and Qemu platform for hob change.

Signed-off-by: Guo Dong <guo.dong@intel.com>

### CODE REVIEW

File: BootloaderCorePkg/Stage2/Stage2Support.c [hunk 1]
```diff
@@ -403,33 +404,26 @@ BuildBaseInfoHob (
     SerialPortInfo->UartPciAddr = 0;
   }
 
-  // Build Frame Buffer hob
-  FspGfxHob = (EFI_PEI_GRAPHICS_INFO_HOB *)GetGuidHobData (LdrGlobal->FspHobList, &Length,
+  // Build graphic info hob
+  BlGfxHob = BuildGuidHob (&gEfiGraphicsInfoHobGuid, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
+  ZeroMem (BlGfxHob, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
```
Comment (mauricema) [inline BootloaderCorePkg/Stage2/Stage2Support.c]: ZeroMem needs to be inside if (blGfxHob != NULL)

File: BootloaderCorePkg/Stage2/Stage2Support.c [hunk 2]
```diff
@@ -403,33 +404,26 @@ BuildBaseInfoHob (
...
 
-  if (FrameBuffer.LinearFrameBuffer) {
-    FrameBufferInfo = BuildGuidHob (&gLoaderFrameBufferInfoGuid, sizeof (FRAME_BUFFER_INFO));
-    CopyMem (FrameBufferInfo, &FrameBuffer, sizeof (FRAME_BUFFER_INFO));
-    DEBUG ((DEBUG_INFO, "Framebuffer Info: %d x %d x 32 @ 0x%08X\n", \
-            FrameBuffer.HorizontalResolution, FrameBuffer.VerticalResolution, FrameBuffer.LinearFrameBuffer));
-  }
+  // Build graphic device info hob
+  BlGfxDeviceInfo = BuildGuidHob (&gEfiGraphicsDeviceInfoHobGuid, sizeof (EFI_PEI_GRAPHICS_DEVICE_INFO_HOB));
+  SetMem (BlGfxDeviceInfo, sizeof (EFI_PEI_GRAPHICS_DEVICE_INFO_HOB), 0xFF);
```
Comment (mauricema) [inline BootloaderCorePkg/Stage2/Stage2Support.c]: why do we set to 0xFF by default ?

File: BootloaderCorePkg/Stage2/Stage2Support.c [hunk 3]
```diff
@@ -403,33 +404,26 @@ BuildBaseInfoHob (
...
   }
 
-  if (FrameBuffer.LinearFrameBuffer) {
-    FrameBufferInfo = BuildGuidHob (&gLoaderFrameBufferInfoGuid, sizeof (FRAME_BUFFER_INFO));
-    CopyMem (FrameBufferInfo, &FrameBuffer, sizeof (FRAME_BUFFER_INFO));
-    DEBUG ((DEBUG_INFO, "Framebuffer Info: %d x %d x 32 @ 0x%08X\n", \
-            FrameBuffer.HorizontalResolution, FrameBuffer.VerticalResolution, FrameBuffer.LinearFrameBuffer));
-  }
+  // Build graphic device info hob
+  BlGfxDeviceInfo = BuildGuidHob (&gEfiGraphicsDeviceInfoHobGuid, sizeof (EFI_PEI_GRAPHICS_DEVICE_INFO_HOB));
```
Comment (mauricema) [inline BootloaderCorePkg/Stage2/Stage2Support.c]: Should we check NULL for BlGfxDeviceInfo?

File: BootloaderCorePkg/Stage2/Stage2Support.c [hunk 4]
```diff
@@ -499,32 +493,33 @@ BuildBaseInfoHob (
 VOID *
 EFIAPI
 BuildExtraInfoHob (
-  IN  STAGE2_HOB  *Stage2Hob
+  IN  STAGE2_HOB                   *Stage2Hob
```
Comment (mauricema) [inline BootloaderCorePkg/Stage2/Stage2Support.c]: can we add function parameter comments?

File: PayloadPkg/OsLoader/Linux.c [hunk 5]
```diff
@@ -244,23 +244,23 @@ SetupBzImage (
...
-    Bp->ScreenInfo.LfbHeight       = (UINT16)FrameBufferInfo->VerticalResolution;
+    Bp->ScreenInfo.LfbLinelength   = (UINT16) (GfxInfoHob->GraphicsMode.PixelsPerScanLine * 4);
+    Bp->ScreenInfo.LfbDepth        = 32;
+    Bp->ScreenInfo.LfbBase         = (UINT32)(UINTN)GfxInfoHob->FrameBufferBase;
+    Bp->ScreenInfo.LfbWidth        = (UINT16)GfxInfoHob->GraphicsMode.HorizontalResolution;
+    Bp->ScreenInfo.LfbHeight       = (UINT16)GfxInfoHob->GraphicsMode.VerticalResolution;
     Bp->ScreenInfo.Pages           = 1;
     Bp->ScreenInfo.BlueSize        = 8;
-    Bp->ScreenInfo.BluePos         = FrameBufferInfo->Blue.Position;
+    Bp->ScreenInfo.BluePos         = 0x00;
```
Comment (mauricema) [inline PayloadPkg/OsLoader/Linux.c]: why do we hardcode ?   we should read it from EFI_PIXEL_BITMASK in HOB.

Comment (mauricema) [reply PayloadPkg/OsLoader/Linux.c]: Same for GreenPos and RedPos.

File: BootloaderCommonPkg/Library/MultibootLib/Multiboot.c [hunk 6]
```diff
@@ -209,21 +209,21 @@ SetupMultibootInfo (
...
-    MbInfo->FramebufferHeight = FrameBufferInfo->VerticalResolution;
-    MbInfo->FramebufferBpp    = (UINT8)FrameBufferInfo->BitsPerPixel;
+    MbInfo->FramebufferAddr   = (UINT64)(UINTN)GfxInfoHob->FrameBufferBase;
+    MbInfo->FramebufferPitch  = GfxInfoHob->GraphicsMode.PixelsPerScanLine * 4;
+    MbInfo->FramebufferWidth  = GfxInfoHob->GraphicsMode.HorizontalResolution;
+    MbInfo->FramebufferHeight = GfxInfoHob->GraphicsMode.VerticalResolution;
+    MbInfo->FramebufferBpp    = 32;
     MbInfo->FramebufferType   = 1;
-    MbInfo->FramebufferRedFieldPosition   = FrameBufferInfo->Red.Position;
+    MbInfo->FramebufferRedFieldPosition   = 0x10;
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/MultibootLib/Multiboot.c]: should not use hardcoded value.

File: BootloaderCorePkg/Stage2/Stage2Support.c [hunk 7]
```diff
@@ -403,33 +404,26 @@ BuildBaseInfoHob (
...
-    FrameBuffer.Green.Position       = 0x08;
-    FrameBuffer.Green.Mask           = 0xFF;
-    FrameBuffer.Blue.Position        = 0x00;
-    FrameBuffer.Blue.Mask            = 0xFF;
-    FrameBuffer.Reserved.Position    = 0x18;
-    FrameBuffer.Reserved.Mask        = 0xFF;
-  } else {
-    FrameBuffer.LinearFrameBuffer    = 0;
+    if (FspGfxHob != NULL) {
+      CopyMem (BlGfxHob, FspGfxHob, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
```
Comment (mauricema) [inline BootloaderCorePkg/Stage2/Stage2Support.c]: Let us verify the EFI_GRAPHICS_PIXEL_FORMAT.
We only support PixelRedGreenBlueReserved8BitPerColor and PixelBlueGreenRedReserved8BitPerColor.

File: BootloaderCorePkg/Stage2/Stage2Support.c (hunk 1 above)
Comment (gdong1) [reply BootloaderCorePkg/Stage2/Stage2Support.c]: thank to capture this. will updated in next update.

File: BootloaderCorePkg/Stage2/Stage2Support.c (hunk 2 above)
Comment (gdong1) [reply BootloaderCorePkg/Stage2/Stage2Support.c]: By the definition of the structure. 0xFF or 0xFFFF will be ignored.

File: BootloaderCorePkg/Stage2/Stage2Support.c (hunk 3 above)
Comment (gdong1) [reply BootloaderCorePkg/Stage2/Stage2Support.c]: will add NULL check.

File: BootloaderCorePkg/Stage2/Stage2Support.c (hunk 4 above)
Comment (gdong1) [reply BootloaderCorePkg/Stage2/Stage2Support.c]: sure. will update function comments.

File: PayloadPkg/OsLoader/Linux.c (hunk 5 above)
Comment (gdong1) [reply PayloadPkg/OsLoader/Linux.c]: All these hardcoded value copied from Stage2Support.c when it builds FB HOB. 
will try to remove hard code value when possible.

File: BootloaderCommonPkg/Library/MultibootLib/Multiboot.c (hunk 6 above)
Comment (gdong1) [reply BootloaderCommonPkg/Library/MultibootLib/Multiboot.c]: same reason as above

File: BootloaderCorePkg/Stage2/Stage2Support.c (hunk 7 above)
Comment (gdong1) [reply BootloaderCorePkg/Stage2/Stage2Support.c]: ok

File: BootloaderCorePkg/Stage2/Stage2Support.c (hunk 2 above)
Comment (mauricema) [reply BootloaderCorePkg/Stage2/Stage2Support.c]: ok

File: PayloadPkg/OsLoader/Linux.c (hunk 5 above)
Comment (mauricema) [reply PayloadPkg/OsLoader/Linux.c]: ok

**Pull request merged**